### PR TITLE
Fix all Interface Builder and compiler warnings

### PR DESCRIPTION
**Pull request by zorgiepoo**
This is aimed to fix/resolve all interface builder and compiler warnings (there was one unused installationPath variable). There were also some minor Xcode project warnings.

To resolve the IB warnings:
* SUStatus.xib now uses auto layout constraints to avoid (localization) ambiguities
* SUPermissionPrompt now uses auto layout constraints to avoid (localization) ambiguities
* SUPermissionPrompt now uses NSStackView to conditionally detach/hide views, specifically when anonymous profile information can be shown and expanded (this works well with auto layout).
  - This removes boilerplate that auto-adjusts the window frame and presently causes a 'glitch' when invoking the disclosure
  - This makes the disclosure button properly bound to the checkbox button in RTL languages like Hebrew 

The last one (along with using automatic preferred width mode on UI elements) requires bumping the minimum requirement to **macOS 10.11 (El Capitan)**. If approved, I'm going to obsolete 10.9 references in our repo ([edit]: I have now replaced all references in this PR)

In the end, the project emits no warnings which positively impacts clients that build Sparkle as a dependency.

## Checklist:

- [x] My change is being tested and reviewed against the Sparkle 2.x branch. New changes must be developed on the 2.x development branch first.
- [ ] My change is being backported to master branch (Sparkle 1.x). Please create a separate pull request for 1.x, should it be backported. Note 1.x is feature frozen and is only taking bug fixes, localization updates, and critical OS adoption enhancements.
- [x] I have reviewed and commented my code, particularly in hard-to-understand areas.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] My change is or requires a documentation or localization update

## Testing

I tested and verified my change by using one or multiple of these methods:

- [x] Sparkle Test App
- [ ] Unit Tests
- [x] My own app
- [ ] Other (please specify)

Tested apps showing permission prompt, with SUEnableSystemProfiling set to 1 and 0 in Info.plist. Tested downloading/installing update. Tested multiple languages (English, Japanese, Hebrew). 

macOS version tested: 11.4 (20F71), latest 10.12 OS in VM (closest working VM I have)

### CODE REVIEW

File: Configurations/ConfigTestApp.xcconfig [hunk 1]
```diff
@@ -8,4 +8,3 @@ PRODUCT_BUNDLE_IDENTIFIER = org.sparkle-project.SparkleTestApp
 CODE_SIGN_ENTITLEMENTS = TestApplication/Sparkle-Test-App.entitlements
 MACOSX_DEPLOYMENT_TARGET = 10.10
```
Comment (tonyarnold) [inline Configurations/ConfigTestApp.xcconfig:9]:
```suggestion
MACOSX_DEPLOYMENT_TARGET = 10.11
```

Comment (zorgiepoo) [reply Configurations/ConfigTestApp.xcconfig]: Good catch. We had several components already with 10.10 or 10.11 min requirement outside of the framework (unit tests, test app, sparkle-cli, ..). I just removed deployment target on all of these and they now inherit 10.11 from the common config.

Comment (tonyarnold) [reply Configurations/ConfigTestApp.xcconfig]: Excellent decision 👍🏻

**Review by kornelski — APPROVED** [review]

**Comment by zorgiepoo** [comment]
I had also re-added the animation when disclosing anonymous profile info in the permission prompt.

**Pull request merged**